src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.cs

```diff
@@ -62,6 +66,79 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                 .WithTrackingName(GenSpecTrackingName);
 
             context.RegisterSourceOutput(genSpec, ReportDiagnosticsAndEmitSource);
+
+            if (!s_hasInitializedInterceptorVersion)
+            {
+                InterceptorVersion = DetermineInterceptableVersion();
+                s_hasInitializedInterceptorVersion = true;
+            }
+        }
+
+        internal static int s_interceptorVersion;
+        internal static int InterceptorVersion
```

Comment:
>InterceptorVersion [](http://example.com/codeflow?start=28&length=18)

nit: why this is not a simple { get; set;}? I see the assert but I am not sure if worth it.

Reply:
Changed - at one point it had additional logic.